docs/production/deployment.md

```diff
@@ -830,9 +830,12 @@ When the [S3 storage backend][s3-backend] is in use, downloads from S3 are
 proxied from nginx, whose configuration requires an explicit value of a DNS
 nameserver to resolve the S3 server's hostname. Zulip defaults to using the
 resolver found in `/etc/resolv.conf`; this setting overrides any value found
-there.
+there. The format should follow the format for [the nginx `resolver`
+directive][nginx-resolver]: IPv6 addresses should be enclosed in brackets, and
+an optional port is allowed.
```

Comment:
Hmm, should we instead take care of this for the user by having the puppet logic do the bracket addition if appropriate for user-provided values too?

Reply:
The heuristic is a bit fiddly because the IPv4 can have a `:` if it has a port, but I think "contains a `:` and no `.`" should work?